lib/client/signalflow/message_router.js

```diff
@@ -80,11 +80,17 @@ function getRoutedMessageHandler(params, onMessage) {
             numBatchesDetermined = true;
             flushBuffer();
           }
+
           onMessage(msg);
           break;
         case 'metadata':
         case 'event':
+          onMessage(msg);
+          break;
         case 'control-message':
+          if (isRetryPatchMode && !numBatchesDetermined) {
```

Comment:
this is the correct logic I believe.  we will only skip this message if we are in patch mode and we have not yet determined the number of messages per batch, which is set by the receipt of JOB_RESOLUTION or any two differing timestamped databatches, meaning end of channel messages will not be skipped because we only do this up to the first iteration in any case.